#Scenes/SceneScripts/credits.gd

```diff
@@ -0,0 +1,327 @@
+	"In/gamer_in_1": 1.5,
+	"In/maid_in_1": 0.7,
+	"In/milf_in_1": 2.6,
+	"In/nerd_in_1": 1.7,
+	"In/old_in_1": 1.5,
+	"In/sleepy_in_1": 0.4,
+	"In/snow_in_1": 0.4,
+}
+
+## Time needed for an animation to free the space at the center of the screen [br]
+## Delay the previous animation by this duration if the time it takes to reach the center is smaller
+## than the time it takes the "out" animation to free the screen center
+const ANIMATION_TIME_OUT_SCREEN_CENTER: Dictionary = {
+	"Out/cool_out_1": 1,
+	"Out/emo_out_1": 2.1,
+	"Out/gamer_out_1": 1.3,
+	"Out/maid_out_1": 0.5,
+	"Out/milf_out_1": 0.6,
+	"Out/nerd_out_1": 0.7,
+	"Out/old_out_1": 0.9,
+	"Out/sleepy_out_1": 2.2,
+	"Out/snow_out_1": 1.1,
+}
+
+## Launch the loop of the scene
+func _ready() -> void:
+	# start by setting up various things
+	black_overlay.modulate.a = 0
+	name_role_label.text = ""
+	name_role_label.add_theme_color_override("font_color", COLOR_FADE_OUT)
+	_animation_loop()
+	
+	await animation_ended
+	
+	# fade to black and go to next scene
+	var tween: Tween = get_tree().create_tween()
+	tween.tween_property(black_overlay, "modulate:a", 1, 3)
+	
+	await tween.finished
+	
+	SceneManager.goto_scene("res://#Scenes/credits_extra.tscn")
+	
+	
+
+## The main animation loop, controls the entry timing of all the saplings [br]
+## This is done by controling the two animation player of the scene
+func _animation_loop() -> void:
+	# a little timer to start the scene
+	var timer: SceneTreeTimer = get_tree().create_timer(1)
+	var team_members_names: Array = _choose_member_order()
+
+	await timer.timeout
+	
+	# play first animation with animation player in
+	# this is needed since the loop also plays the out animation of the previous avatar
+	# but there is no previous avatar before the first one
+	var first_member: String = team_members_names[0]
+	var roles_and_avatar: Array = TEAM_MEMBERS[first_member]
+	var roles: Array = roles_and_avatar[0]
+	var roles_string: String = _role_array_to_string(roles)
+	var previous_avatar: GlobalEnums.SaplingType = roles_and_avatar[1]
+	# randomly select an avatar type for those that don't have one
+	# change to defaulting to MILF later
+	if previous_avatar == sapling_type.None :
+		previous_avatar = _default_avatar()
+	var animation_name: String = _choose_animation(previous_avatar, true)
+	anim_in.play(animation_name)
+	await anim_in.animation_finished
+	# member is now at the center of the screen, display their name + roles with a fade in of the text
+	name_role_label.text = first_member + "\n" + roles_string
+	var text_time_on_screen: float = _get_text_time_on_screen(name_role_label.text)
+	var tween: Tween = get_tree().create_tween()
+	tween.tween_property(name_role_label, "theme_override_colors/font_color", COLOR_FADE_IN, TEXT_FADE_IN_DURATION)
+	await get_tree().create_timer(text_time_on_screen).timeout
+	
+	var out_animation: String
+	
+	# now loop on all the remaining members
```

Comment:
Is there a reason you do out anim first and then in anims?
The way you have it now means you kinda wrote the same code twice so I think it would be better if it would be possible to make in anims come first and outs come second and then remove the repeat code

Reply:
I'll check the code, maybe there was a reason for the animation in itself

Reply:
The reason we do out anim first then in anim is due the way things are handled in the animation, a little explanation:

We have two players, one that does the animation for the sprites going in, one for the sprites going out.

There is a sprite animated for the animation in, and another sprite animated for the animation out, even if it's the same image. Because if we use the same sprite to do in and out animation, then we can't have the same image going in and out at the same time. For example, if two "Cool" sapling follow each other, with a single sprite we would need to same sprite to both play the in and out animation at the same time, which is not possible, thus the separation.

Now doing it like this forces us to play the out animation before the in animation. What would happen if I played the in before the out:
1. I play the in, so I reset the position of the last sprite that got in (which was at the center of the screen) in order to play the animation of the next sprite that goes in
2. Since I reset the position, there is no more sprite at the center of the screen
3. No I play the out animation, the sprite for the out animation pops at the center of the screen

Note that steps 1-3 are done one after the other very fast. So maybe it wouldn't make a flicker on the sprite, but playing the out before doing a reset on the position of the in ensures that there is no flicker. I could invert it and check that it still works, but on slower computer that might not be the case.

Reply:
I don't think you need to change it then. It's only a small thing, and the system is so isolated that I think it probably fine to leave it like it is.

Reply:
yeah, it probably won't be changed a lot, it doesn't interact with outside stuff anyway
there might be a way to change the code to make it work, but i would have to check